frontend/javascripts/viewer/model/sagas/volume/proofread_saga.ts

```diff
@@ -586,6 +621,159 @@ function* performMinCut(
   return false;
 }
 
+function* performPartitionedMinCut(_action: MinCutPartitionsAction | EnterAction): Saga<void> {
+  const isMultiSplitActive = yield* select((state) => state.userConfiguration.isMultiSplitActive);
+  if (!isMultiSplitActive) {
+    return;
+  }
+
+  const preparation = yield* call(prepareSplitOrMerge, false);
+  if (!preparation) {
+    return;
+  }
+  const partitions = yield* select(
+    (state) => state.localSegmentationData[preparation.volumeTracing.tracingId].minCutPartitions,
+  );
+  if (partitions[1].length <= 0 || partitions[2].length <= 0) {
+    Toast.error(
+      "Not every partition has at least one selected segment. Select at least one segment for each partition before performing a cut action.",
+    );
+    return;
+  }
+  const agglomerateId = partitions.agglomerateId;
+  const volumeTracingId = preparation.volumeTracing.tracingId;
+  const { agglomerateFileMag: mag, activeMapping } = preparation;
+  const agglomerate = preparation.volumeTracing.segments.getNullable(Number(agglomerateId));
+
+  const tracingStoreUrl = yield* select((state) => state.annotation.tracingStore.url);
+  const segmentsInfo = {
+    partitionOne: partitions[1],
+    partitionTwo: partitions[2],
+    mag,
+    agglomerateId: agglomerateId,
+    editableMappingId: preparation.volumeTracing.tracingId,
+  };
+
+  let edgesToRemove;
+  try {
+    edgesToRemove = yield* call(
+      getEdgesForAgglomerateMinCut,
+      tracingStoreUrl,
+      volumeTracingId,
+      segmentsInfo,
+    );
+  } catch (exception) {
+    console.error(exception);
+    Toast.error("Could not determine which edges to delete for cut. Please try again.");
+    return;
+  }
+  if (edgesToRemove.length <= 0) {
+    Toast.error(
+      "Could not detect any edges to split to perform the desired cut operation. Please try again.",
+    );
+  }
+  const items = [];
+
+  for (const edge of edgesToRemove) {
+    console.log(
+      "Splitting agglomerate",
+      agglomerateId,
+      "with segment ids",
+      edge.segmentId1,
+      "and",
+      edge.segmentId2,
+    );
+    items.push(
+      splitAgglomerate(agglomerateId, edge.segmentId1, edge.segmentId2, mag, volumeTracingId),
+    );
+  }
+
+  yield* put(pushSaveQueueTransaction(items));
+  yield* call([Model, Model.ensureSavedState]);
+  yield* put(resetMultiCutToolPartitionsAction());
+
+  console.log("start updating the mapping after a min-cut");
+  const unmappedSegmentsOfPartitions = [...partitions[1], ...partitions[2]];
+  // Make sure the reloaded partial mapping has mapping info about the partitions and first removed edge. The first removed edge is used for reloading the meshes.
```

Comment:
I don't understand why the first removed edge is important :thinking: Could you explain that?